viewer/src/containers/Search/saga.tsx

```diff
@@ -78,18 +118,26 @@ function* getDownloadType({ resourceType, params }: any) {
 
       // update the counter
       yield put(downloadBundleUpdateAction(entries.length / count));
-      page += 1;
     } while (entries.length < count);
 
     // put into package
     const download = '['.concat(entries.join(',\n'), ']');
     yield put(downloadBundleSuccessAction(download));
   } catch (e) {
-    downloadBundleErrorAction(e);
+    yield put(downloadBundleUpdateAction(-1));
+    yield put(downloadBundleErrorAction(e));
   }
 }
 
 export default function* searchSaga() {
-  yield all([takeEvery(GET_BUNDLE, getResourceType)]);
-  yield all([takeEvery(GET_DOWNLOAD, getDownloadType)]);
+  yield all([takeEvery(GET_BUNDLE, getBundle)]);
+  yield all([takeEvery(GET_ENTRY, getEntry)]);
+  yield all([takeEvery(GET_DOWNLOAD, getDownload)]);
 }
+
+// util functions
+const makeParamString = (params: any): string => {
+  return Object.entries(params)
+    .map(([k, v]) => `${k}=${v}`)
```

Comment:
Should wrap the value in a URLEncode to be safe.`.map(([k, v]) => `${k}=${encodeURIComponent(v)}`)`

Reply:
fixed in adcecb88711f644bcf3a7abcda15469a13100ba6